apps/remote_control/test/lexical/remote_control/code_intelligence/definition_test.exs

```diff
@@ -0,0 +1,370 @@
+defmodule Lexical.RemoteControl.CodeIntelligence.DefinitionTest do
+  alias Lexical.RemoteControl.CodeIntelligence.Definition
+  alias Lexical.SourceFile
+  alias Lexical.SourceFile.Line
+  alias Lexical.SourceFile.Position
+
+  import Lexical.Test.Fixtures
+  import Lexical.Test.CodeSigil
+  import Line
+
+  use ExUnit.Case, async: false
+
+  setup do
+    start_supervised!(Lexical.SourceFile.Store)
+    Code.put_compiler_option(:ignore_module_conflict, true)
+    :ok
+  end
+
+  defp with_navigation_project(_ctx) do
+    %{project: project(:navigations)}
+  end
+
+  defp with_referenced_file(%{project: project}, file \\ "my_definition.ex") do
+    path = file_path(project, Path.join("lib", file))
+    Code.compile_file(path)
+    %{uri: SourceFile.Path.ensure_uri(path)}
+  end
+
+  defp open_uses_file(project, content) do
+    uri =
+      project
+      |> file_path(Path.join("lib", "my_module.ex"))
+      |> SourceFile.Path.ensure_uri()
+
+    with :ok <- SourceFile.Store.open(uri, content, 1),
+         {:ok, uses_file} <- SourceFile.Store.fetch(uri) do
+      uses_file
+    end
+  end
+
+  defp cursor_to_position(cursor, source_file) do
+    find_line = fn source_file, cursor ->
+      Enum.find(Tuple.to_list(source_file.document.lines), fn line(text: text) ->
+        cursor_full_text = String.replace(cursor, "|", "")
+        String.contains?(text, cursor_full_text)
+      end)
+    end
+
+    line(line_number: line_number) = find_line.(source_file, cursor)
+    {:ok, text} = SourceFile.fetch_text_at(source_file, line_number)
+    [cursor | _] = String.split(cursor, "|", parts: 2)
+    [before_cursor, _] = String.split(text, cursor, parts: 2)
+    character = String.length(before_cursor) + String.length(cursor) + 1
+    Position.new(line_number, character)
```

Comment:
all this duplicates a lot of what you get in `CursorSupport`

Reply:
Removed